classes/class-gf-securesubmit.php

```diff
@@ -1,950 +1,1597 @@
+                    'updateAuthorizationEntry'],
+                   10,
+                   2);
+        add_filter('gform_replace_merge_tags',
+                   [$this,
+                    'replaceMergeTags'],
+                   10,
+                   7);
+        add_action('gform_admin_pre_render',
+                   [$this,
+                    'addClientSideMergeTags']);
+
+        /* Sets WP to call \GFSecureSubmit::hps_add_cc_field and build our button
+        src: wordpress/wp-includes/plugin.php
+        * \GFSecureSubmit::hps_add_cc_field
+        * * Sets WP to call \GFSecureSubmit::hps_add_ach_field and build our button
+         src: wordpress/wp-includes/plugin.php
+        * \GFSecureSubmit::hps_add_ach_field
+        */
+        add_filter('gform_add_field_buttons',
+                   [$this,
+                    'hps_add_ach_field']);
+        add_action( 'gform_editor_js_set_default_values', [$this,'set_defaults'] );
+    }
+
+    function set_defaults(){
+        ?>
+        //this hook is fired in the middle of a switch statement,
+        //so we need to add a case for our new field type
+        case "hpsACH" :
+        field.label = "Bank Transfer"; //setting the default field label
+        break;
+        <?php //////
+    }
+
+    /**
+     *
+     */
+    public
+    function init_ajax()
+    {
+        parent::init_ajax();
+        add_action('wp_ajax_gf_validate_secret_api_key',
+                   [$this,
+                    'ajaxValidateSecretApiKey']);
+    }
+
+    /**
+     *
+     */
+    public
+    function ajaxValidateSecretApiKey()
+    {
+        $this->includeSecureSubmitSDK();
+        $config                = new HpsServicesConfig();
+        $config->secretApiKey  = rgpost('key');
+        $config->developerId   = '002914';
+        $config->versionNumber = '1916';
+
+        $service = new HpsCreditService($config);
+
+        $is_valid = true;
+
+        try {
+            $service->get('1');
+        }
+        catch (HpsAuthenticationException $e) {
+            $is_valid = false;
+        }
+        catch (HpsException $e) {
+            // Transaction was authenticated, but failed for another reason
+        }
+
+        $response = $is_valid
+            ? 'valid'
+            : 'invalid';
+        die($response);
+    }
+
+    /**
+     * @return array
+     */
+    public
+    function plugin_settings_fields()
+    {
+        return [['title'  => __('SecureSubmit API', $this->_slug),
+                 'fields' => $this->sdkSettingsFields(),],
+                ['title'  => __('Velocity Limits', $this->_slug),
+                 'fields' => $this->vmcSettingsFields(),],];
+    }
+
+    /**
+     * @return bool|false|string
+     */
+    public
+    function feed_list_message()
+    {
+
+        // if ( $this->_requires_credit_card && ! $this->has_credit_card_field( $this->get_current_form() ) ) {
+        //     return $this->requires_credit_card_message();
+        // }
```

Comment:
The messaging here should remain in play. The fee should check if there is an appropriate field type currently in the form.